File: App/Dashboard/Dashboard.DataAccess/Models/Entities/Orders/OrderDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Dashboard.DataAccess.Models.Entities.Base;
using Dashboard.DataAccess.Models.Entities.Products;
using Microsoft.EntityFrameworkCore;

namespace Dashboard.DataAccess.Models.Entities.Orders;

[Table("order_details")]
public partial class OrderDetail : BaseAuditableEntity
{
    [Column("quantity")]
    public int Quantity { get; set; }

    [Column("order_id")]
    public long OrderId { get; set; }

    [Column("product_id")]
    public long ProductId { get; set; }

    [Column("color")]
    [StringLength(255)]
    [Unicode(false)]
    public string? Color { get; set; }

    [Column("note")]
    [StringLength(255)]
    [Unicode(false)]
    public string? Note { get; set; }

    [Column("total_amount", TypeName = "decimal(18, 2)")]
    public decimal TotalAmount { get; set; }

    [Column("unit_price", TypeName = "decimal(18, 2)")]
    public decimal UnitPrice { get; set; }

    [ForeignKey("OrderId")]
    [InverseProperty("OrderDetails")]
    public virtual Order Order { get; set; } = null!;

    [ForeignKey("ProductId")]
    [InverseProperty("OrderDetails")]
    public virtual Product Product { get; set; } = null!;
}

File: App/backEnd/src/main/java/com/project/webbanhang/models/orders/OrderDetail.java
package com.project.webbanhang.models.orders;

import com.project.webbanhang.models.BaseEntity;
import com.project.webbanhang.models.Product;
import jakarta.persistence.*;
import lombok.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
@Builder
@Table(name = "order_details")
public class OrderDetail extends BaseEntity {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "order_id", nullable = false)
	private Order order;

	@Column(name = "quantity")
	private Long quantity;

	@ManyToOne
	@JoinColumn(name = "product_id", nullable = false)
	private Product product;

	@Column(name = "size")
	private String size;

	@Column(name = "note")
	private String note;

	@Column(name = "total_amount")
	private Long totalAmount;

	@Column(name = "unit_price")
	private Long unitPrice;
}
